torch/csrc/inductor/aoti_torch/c/shim.h

```diff
@@ -152,6 +152,22 @@ AOTI_TORCH_EXPORT AOTITorchError aoti_torch_create_tensor_from_blob(
     AtenTensorHandle* ret // returns new reference
 );
 
+AOTI_TORCH_EXPORT AOTITorchError aoti_torch__scaled_dot_product_flash_attention(
+    AtenTensorHandle* ret0, // returns new reference
+    AtenTensorHandle* ret1, // returns new reference
+    AtenTensorHandle* ret2, // returns new reference
+    AtenTensorHandle* ret3, // returns new reference
+    int64_t* ret4,
+    int64_t* ret5,
+    AtenTensorHandle* ret6, // returns new reference
+    AtenTensorHandle* ret7, // returns new reference
+    AtenTensorHandle* ret8, // returns new reference
+    int32_t num_inputs,
+    AtenTensorHandle query,
+    AtenTensorHandle key,
+    AtenTensorHandle value,
+    ...);
```

Comment:
Variadic function in the ABI?! Seems suspicious. Tell me more?

Reply:
Agree, looks kinda suspicious. Maybe we need to define something like:

```
aoti_torch__scaled_dot_product_flash_attention_6()
aoti_torch__scaled_dot_product_flash_attention_7()
aoti_torch__scaled_dot_product_flash_attention_8()
```

Reply:
Naively, I would have expected to take all optional arguments and force you to fill in the defaults.

Reply:
```torch.ops.aten._scaled_dot_product_flash_attention``` takes a couple of default arguments (dropout_p, is_causal, return_debug_mask and scale) that may not be presented in the IR. For example, given ``` torch.ops.aten._scaled_dot_product_flash_attention(q, k, v)```, at https://github.com/pytorch/pytorch/blob/main/torch/_inductor/ir.py#L3853, we only see three args for ```q```, ```k```, ```v```. We are missing all the default arguments. Consequently, at the codegen time, we don't know the default values for generating ```_scaled_doc_product_flash_attention``` fallback. (Question - is there any way that we could get the default argument values when we create IR for this FallbackKernel? )

Note that we could make default values for ```_scaled_doc_product_flash_attention``` by explicitly detecting the kernel name in the wrapper code, but I think it may be not general enough. For example, we might end up with a number of special rules for various fallback kernels in the wrapper. 

Since C doesn't support default arguments, I went with the variadic-function solution.

Reply:
`ordered_kwargs_for_cpp_kernel` should give you the default values for which is read from schema.

Reply:
> `ordered_kwargs_for_cpp_kernel` should give you the default values for which is read from schema.

Hmm, we don't use schema for ```_scaled_dot_product_flash_attention```. Do we want to use schema for it in a way similar to other kernels like ```ConvolutionUnary```?

Reply:
> Agree, looks kinda suspicious. Maybe we need to define something like:
> 
> ```
> aoti_torch__scaled_dot_product_flash_attention_6()
> aoti_torch__scaled_dot_product_flash_attention_7()
> aoti_torch__scaled_dot_product_flash_attention_8()
> ```

Hmm, it's probably less ideal, because we would have a special rule for the ```scaled_dot_product_flash_attention``` fallback in the wrapper code to generate these.

Reply:
> Naively, I would have expected to take all optional arguments and force you to fill in the defaults.

Yeah, makes sense.

Reply:
Per discussion, replaced the variadic function with the one with default arguments being filled up. This change requires us to replace ```torch.ops.aten._scaled_dot_product_flash_attention``` with ```torch.nn.functional.scaled_dot_product_attention```. Only the latter provides a schema that can be used to retrieve the default values for the optional arguments.